src/content/en/updates/2018/04/loading-wasm.md

```diff
@@ -0,0 +1,158 @@
+
+{# wf_updated_on: 2018-04-12 #}
+{# wf_published_on: 2018-04-12 #}
+{# wf_tags: javascript, webassembly #}
+{# wf_featured_image: /web/updates/images/generic/timer.png #}
+{# wf_featured_snippet: When working with WebAssembly, you often want to download a module, compile it, instantiate it, and then use whatever it exports in JavaScript. This post explains our recommended approach for optimal efficiency. #}
+{# wf_blink_components: Blink>JavaScript #}
+
+# Loading WebAssembly modules efficiently {: .page-title }
+
+{% include "web/_shared/contributors/mathiasbynens.html" %}
+
+When working with WebAssembly, you often want to download a module, compile it, instantiate it, and
+then use whatever it exports in JavaScript. This post starts off with a common but suboptimal code
+snippet doing exactly that, discusses several possible optimizations, and eventually shows the
+simplest, most efficient way of running WebAssembly from JavaScript.
+
+Note: Tools like Emscripten can produce the needed boilerplate code for you, so you don’t
+necessarily have to code this yourself. In cases where you need fine-grained control over the
+loading of WebAssembly modules however, it helps to keep the following best practices in mind.
+
+This code snippet does the complete download-compile-instantiate dance, albeit in a suboptimal way:
+
+```js
+// Don’t use this!
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const buffer = await response.arrayBuffer();
+  const module = new WebAssembly.Module(buffer);
+  const instance = new WebAssembly.Instance(module);
+  const result = instance.exports.fibonacci(42);
+})();
+```
+
+Note how we use `new WebAssembly.Module(buffer)` to turn a response buffer into a module. This is a
+synchronous API, meaning it blocks the main thread until it completes. To discourage its use, Chrome
+disables `WebAssembly.Module` for buffers larger than 4 KB. To work around the size limit, we can
+use `await WebAssembly.compile(buffer)` instead:
+
+```js
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const buffer = await response.arrayBuffer();
+  const module = await WebAssembly.compile(buffer);
+  const instance = new WebAssembly.Instance(module);
+  const result = instance.exports.fibonacci(42);
+})();
+```
+
+`await WebAssembly.compile(buffer)` is _still_ not the optimal approach, but we’ll get to that in a
+second.
+
+Almost every operation in the modified snippet is now asynchronous, as the use of `await` makes
+clear. The only exception is `new WebAssembly.Instance(module)`. For consistency, we can use the
+asynchronous `WebAssembly.instantiate(module)`.
+
+```js
+(async () => {
+  const response = await fetch('fibonacci.wasm');
+  const buffer = await response.arrayBuffer();
+  const module = await WebAssembly.compile(buffer);
+  const instance = await WebAssembly.instantiate(module);
+  const result = instance.exports.fibonacci(42);
```

Comment:
How about simply `console.log(instance.exports.fibonacci(42))`; to make it clear when it's finished when executed in console?